### PR TITLE
Fixes #26618 - show cancelled report generating

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Adds an more user friendly information about cancellation of report generating.

message

Tests needed (anticipated impacts):
* Report templates generating in UI / API

### CODE REVIEW

File: app/controllers/api/v2/report_templates_controller.rb [hunk 1]
```diff
@@ -173,7 +173,9 @@ def report_data
         if @plan.progress < 1
           head :no_content
         elsif @plan.failure?
-          render json: { errors: @plan.errors }, status: :unprocessable_entity
+          errors = [{ message: _('Report generating has been canceled') }] if @plan.result == :cancelled
```
Comment (ares) [inline app/controllers/api/v2/report_templates_controller.rb]: `Generating of the report has been cancelled.` maybe? Otherwise LGTM

**Review by ares — CHANGES_REQUESTED** [review]

File: app/controllers/api/v2/report_templates_controller.rb (hunk 1 above)
Comment (ezr-ondrej) [reply app/controllers/api/v2/report_templates_controller.rb]: :dart: yeah, sounds better.

Comment (ekohl) [reply app/controllers/api/v2/report_templates_controller.rb]: Could also be `Report generation was canceled`. Not a native speaker, but I think even `Report generation canceled` could work.

Comment (ezr-ondrej) [reply app/controllers/api/v2/report_templates_controller.rb]: I have used that, but @ares pointed out - quite right - `generation` is a misleading to population (parents, children)

**Review by ares — APPROVED** [review]

**Review by ares — APPROVED** [review]
Works nicely, merging, thanks @ezr-ondrej

**Pull request merged**